lib/rbi/rewriters/replace_attributes_with_methods.rb

```diff
@@ -0,0 +1,146 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RBI
+  module TempHelpers
+    extend T::Sig
+  end
+
+  module Rewriters
+    class ReplaceAttributesWithMethods < Visitor
+      extend T::Sig
+      include TempHelpers
+
+      sig { override.params(node: T.nilable(Node)).void }
+      def visit(node)
+        return unless node
+
+        case node
+        when Tree
+          node.nodes.dup.each do |child|
+            visit(child)
+            next unless (attr = child).is_a?(Attr)
+
+            new_methods = convert_to_methods(attr)
+
+            child.replace_with_multiple(new_methods)
+          end
+        end
+      end
+
+      private
+
+      sig { params(attr: Attr).returns(T::Array[Method]) }
+      def convert_to_methods(attr)
+        sig, attribute_type = parse_sig_of(attr)
```

Comment:
We could move this to where we actually create the resulting methods and let them `translate` the sig rather than passing it around.

* going from `attr_reader :a` to `def a; end` shouldn't require any sig translation
* going from `attr_writer :a` to `def a=(val); end` may require a sig translation
* going from `attr_accessor :a` requires a sig translation only for the writer